host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1448,7 +1594,30 @@ static MMAL_STATUS_T create_camera_component(RASPIVID_STATE *state)
    // Set the encode format on the video  port
 
    format = video_port->format;
-   format->encoding_variant = MMAL_ENCODING_I420;
+
+   if (state->raw_output)
+   {
+      switch (state->raw_output_fmt)
```

Comment:
Unless I've missed something, this is not going to be great.
Video_splitter should be able to do the format conversions (if not then that needs fixing), so significantly better to keep this connection as ENCODING_OPAQUE, and do the conversion on the one output of video_splitter. Otherwise in the RGB case camera will be converting from YUV to RGB, and video_encode will be converting back again as it requires YUV.

I would actually recommend putting the splitter on the preview output (camera->output[0]) and leave output[1] going to the encoder as ENCODING_OPAQUE. Due to various historic reasons, camera->output[1] is optimised for video_encode and provides extra data that makes the encode more efficient. video_splitter doesn't support those optimisations, so inserting it between the two breaks part of the encode path.